plutus-contract/src/Plutus/Contract/Test/DynamicLogic/Monad.hs

```diff
@@ -0,0 +1,116 @@
+{-# LANGUAGE FlexibleContexts #-}
+module Plutus.Contract.Test.DynamicLogic.Monad
+    ( DL
+    , action
+    , anyAction
+    , anyActions
+    , anyActions_
+    , stopping
+    , weight
+    , getModelStateDL
+    , assert
+    , assertModel
+    , monitorDL
+    , forAllQ
+    , forAllDL
+    , forAllMappedDL
+    , withDLTest
+    , DL.DynLogic
+    , DL.DynLogicModel(..)
+    , DL.DynLogicTest(..)
+    , DL.TestStep(..)
+    , module Plutus.Contract.Test.DynamicLogic.Quantify
+    ) where
+
+import           Control.Applicative
+import           Control.Monad
+import           Data.Typeable
+
+import qualified Plutus.Contract.Test.DynamicLogic          as DL
+import           Plutus.Contract.Test.DynamicLogic.Quantify
+import           Plutus.Contract.Test.StateModel
+
+import           Test.QuickCheck
+
+newtype DL s a = DL { unDL :: s -> (a -> DL.DynPred s) -> DL.DynLogic s }
```

Comment:
Assuming we're going to maintain this stuff ourselves, it would be very helpful to have some design comments on the internal bits, even if they're not in the API. That applies throughout, ideally.